models/agent.go

```diff
@@ -211,3 +212,40 @@ func (q *QanAgent) DSN(service *MySQLService) string {
 	// TODO Other parameters?
 	return cfg.FormatDSN()
 }
+
+// AgentFilters represents filters for agents list.
+type AgentFilters struct {
+	ServiceID *string
+}
+
+// AgentsByFilters returns agents providing insights for a given filters.
+func AgentsByFilters(q *reform.Querier, filters AgentFilters) ([]*AgentRow, error) {
+	var agentIDs []interface{}
+	var structs []reform.Struct
+	var err error
+	if filters.ServiceID != nil {
+		agentServices, err := q.SelectAllFrom(AgentServiceView, "WHERE service_id = ? ORDER BY ID", *filters.ServiceID)
```

Comment:
There is no `ID` column.

That code branch is not covered by tests.

Reply:
Sorry, but that code can't work: there is no `ID` column in `agent_services` table. The only conclusion is that tests are not run or not testing that branch.

Reply:
tests worked and failed. https://travis-ci.org/percona/pmm-managed/jobs/480284133